doc/architectural/compilation-and-development.md

```diff
@@ -270,3 +270,28 @@ branch, e.g. to run it on lines that have been changed from `develop`:
 
 There are also instructions for adding this as a git pre-commit hook in
 [CODING_STANDARD.md](https://github.com/diffblue/cbmc/blob/develop/CODING_STANDARD.md#pre-commit-hook-to-run-cpplint-locally).
+
+
+\section compilation-and-development-section-time-profiling Time profiling
+
+To do time profiling with a tool like `gprof`, the flags `-g` and `-pg` must be
+used which compiling, and `-pg` must be used when linking. If you are building
+with cmake you can just add "-Denable_profiling=1" to your cmake invocation, and
```

Comment:
:bulb: I observed on my machine that if I build with cmake twice, first without this option, then with this option (without making any changes to files in between), the second build takes a long time even though I'm using ccache. This might be worth mentioning here.
:grey_question: Is the `=1` necessary?

Reply:
I checked and cmake prints an error without the `=1`